src/renderer/components/+catalog/catalog-add-button.tsx

```diff
@@ -0,0 +1,76 @@
+import "./catalog-add-button.scss";
+import React from "react";
+import { SpeedDial, SpeedDialAction } from "@material-ui/lab";
+import { Icon } from "../icon";
+import { disposeOnUnmount, observer } from "mobx-react";
+import { observable, reaction } from "mobx";
+import { autobind } from "../../../common/utils";
+import { CatalogCategory, CatalogEntityAddMenuContext, CatalogEntityContextMenu } from "../../api/catalog-entity";
+import { EventEmitter } from "events";
+import { navigate } from "../../navigation";
+
+export type CatalogAddButtonProps = {
+  category: CatalogCategory
+};
+
+@observer
+export class CatalogAddButton extends React.Component<CatalogAddButtonProps> {
+  @observable protected isOpen = false;
+  protected menuItems = observable.array<CatalogEntityContextMenu>([]);
+
+  componentDidMount() {
+    const { category } = this.props;
+
+    disposeOnUnmount(this, [
+      reaction(() => category, (category) => {
+        this.menuItems.clear();
+
+        if (category && category instanceof EventEmitter) {
+          const context: CatalogEntityAddMenuContext = {
+            navigate: (url: string) => navigate(url),
+            menuItems: this.menuItems
+          };
+
+          category.emit("onCatalogAddMenu", context);
+        }
+      }, { fireImmediately: true })
+    ]);
+  }
+
+  @autobind()
+  onOpen() {
+    this.isOpen = true;
+  }
+
+  @autobind()
+  onClose() {
+    this.isOpen = false;
+  }
+
+  render() {
+    if (this.menuItems.length === 0) {
+      return null;
+    }
+
+    return (
+      <SpeedDial
+        className="CatalogAddButton"
+        ariaLabel="SpeedDial CatalogAddButton"
+        open={this.isOpen}
+        onOpen={this.onOpen}
+        onClose={this.onClose}
+        icon={<Icon material="add" />}
+        direction="up"
+      >
+        { this.menuItems.map((menuItem, index) => {
+          return <SpeedDialAction
+            key={index}
+            icon={<Icon material="text_snippet" />}
+            tooltipTitle="Import Kubeconfig"
+            onClick={() => menuItem.onClick()}
```

Comment:
Given that menu items can ask for confirmation, shouldn't we support that here?

Reply:
Can you elaborate please? What confirmation should be supported?

Reply:
See what is done on `src/renderer/components/+catalog/catalog.tsx:83` and `src/renderer/components/hotbar/hotbar-icon.tsx:72`.

I think that this is a good reason for having a registry that maps from `CatalogEntityContextMenu` to `ResolvedCatalogEntityContextMenu` so that the code here works as intended.

Reply:
IMHO a bit theoretical use case, we can open an issue about confirmation and implement it later (if there is demand).